ipn/ipnlocal/local.go

```diff
@@ -3682,6 +3682,21 @@ func (b *LocalBackend) resetControlClientLockedAsync() {
 	if b.cc == nil {
 		return
 	}
+
+	// When we clear the control client, stop any outstanding netmap expiry
+	// timer; synthesizing a new netmap while we don't have a control
+	// client will break things.
+	//
+	// See https://github.com/tailscale/tailscale/issues/7392
+	if b.nmExpiryTimer != nil {
+		b.nmExpiryTimer.Stop()
+		b.nmExpiryTimer = nil
+
+		// Also bump the epoch to ensure that if the timer started, it
+		// will abort.
+		b.numClientStatusCalls.Add(1)
```

Comment:
Maybe reset to 0 instead?

Reply:
I thought about it, but I wasn't convinced that there wasn't some sort of race there. It's an `atomic.Uint32`, so it'll wrap around on overflow anyway, so we don't really need to worry about that particular problem.